python/taulabs/telemetry.py

```diff
@@ -323,13 +323,19 @@ def _do_io(self, finish_time):
         if r:
             # Shouldn't throw an exception-- they just told us
             # it was ready for read.
-            chunk = os.read(self.fd, 1024)
-            if chunk == '':
-                raise RuntimeError("stream closed")
+            try:
```

Comment:
I know we don't like TODO comments but I hate the idea of blocking this PR / a useful fix on this anomaly.  Maybe put TODO: in front of the above comment and call it good.